### PR TITLE
Allow multiple "--command" in runTheMatrix

### PR DESCRIPTION
**Pull request by srimanob**
#### PR description:
As discussed in https://github.com/cms-sw/cmssw/pull/34599#issuecomment-890847652
This PR considers `--command` as a list, and join them together. This allows multiple --command(s) to be used in the PR test.

#### PR validation:
`runTheMatrix.py --what upgrade -l 34634.0 --wm init --command '--customise Validation/Performance/TimeMemoryJobReport.customiseWithTimeMemoryJobReport' --command '--procModifiers fineCalo'`
gives the expected output.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:
Not a backport, and no need of backport.

### CODE REVIEW

**Comment by perrotta** [comment]
@smuzaffar

File: Configuration/PyReleaseValidation/python/MatrixReader.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def __init__(self, opt):
 
         self.wm=opt.wmcontrol
         self.revertDqmio=opt.revertDqmio
-        self.addCommand=opt.command
+        self.addCommand=' '.join(opt.command)
```
Comment (smuzaffar) [inline Configuration/PyReleaseValidation/python/MatrixReader.py:25]: @srimanob , by default opt.command is `None` and the above will fail if `--command` option is not used. I would suggest to revert this change and instead add the following
```
    if opt.command: opt.command = ' '.join(opt.command)
```
after https://github.com/cms-sw/cmssw/blob/master/Configuration/PyReleaseValidation/scripts/runTheMatrix.py#L307

**Comment by srimanob** [comment]
test parameters:

- relvals_opt = --command '--customise SLHCUpgradeSimulations/Configuration/aging.customise_aging_3000'

**Comment by srimanob** [comment]
test parameters:

- relvals_opt = --command '--procModifiers fineCalo'

File: Configuration/PyReleaseValidation/python/MatrixReader.py (hunk 1 above)
Comment (srimanob) [reply Configuration/PyReleaseValidation/python/MatrixReader.py]: Thanks @smuzaffar 
I've updated the PR.

**Comment by smuzaffar** [comment]
@srimanob , `/` was not recognized by bot that is why https://github.com/cms-sw/cmssw/pull/34722#issuecomment-891004312 was rejected. I have updated bot to recognize `/` too ( https://github.com/cms-sw/cms-bot/pull/1604 ) . Feel free to try `relvals_opt = --command '--customise SLHCUpgradeSimulations/Configuration/aging.customise_aging_3000'` now

**Comment by srimanob** [comment]
test parameters:

**Comment by srimanob** [comment]
test parameters:

- relvals_opt = --command '--customise SLHCUpgradeSimulations/Configuration/aging.customise_aging_3000'

**Comment by srimanob** [comment]
@cmsbuild please test

Thanks @smuzaffar

**Comment by srimanob** [comment]
Please test

**Comment by smuzaffar** [comment]
please test

**Comment by srimanob** [comment]
Please test

**Comment by srimanob** [comment]
@smuzaffar @perrotta 
The relvals timeout after 4 hours is something to ignore for now? Or we keep testing?

**Comment by srimanob** [comment]
Please test

**Comment by perrotta** [comment]
> @smuzaffar @perrotta
> The relvals timeout after 4 hours is something to ignore for now? Or we keep testing?

RelVals-INPUT timeout affect basically all PR in test. Re-testing will hardly help.
Just ignore them, and stick on all other tests and comparisons for your evaluation.

**Comment by smuzaffar** [comment]
Beside the Relval-INPUT, the normal relvals are not finishing within 4 hours limit. I think the extra `--customise SLHCUpgradeSimulations/Configuration/aging.customise_aging_3000` is slowing down the process. @srimanob do you know what this extra customization does?

**Comment by srimanob** [comment]
Basically, it touches the DIGI step of HCAL, ECAL, HGCAL, HFNose. I have no idea if it increases MC time significantly. I can abort and try another one (aging_1000) as we usually do for upgrade workflows.

**Comment by srimanob** [comment]
Please abort

**Comment by srimanob** [comment]
test parameters:

**Comment by srimanob** [comment]
test parameters:
- relvals_opt = --command '--customise SLHCUpgradeSimulations/Configuration/aging.customise_aging_1000'

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by boudoul** [comment]
> > @smuzaffar @perrotta
> > The relvals timeout after 4 hours is something to ignore for now? Or we keep testing?
> 
> RelVals-INPUT timeout affect basically all PR in test. Re-testing will hardly help.
> Just ignore them, and stick on all other tests and comparisons for your evaluation.

Hello @perrotta - Can you update me on the actions taken to fix this  RelVals-INPUT errors affecting massively the PR tests ? Sorry If I missed information, catching up. Thanks

**Comment by perrotta** [comment]
@boudoul RelVal-INPUT issue is related to #34633
As far as I can see, there is not yet a solution coded in a PR...

**Comment by srimanob** [comment]
Hi @smuzaffar 
How can I spot the fail relvals without going through them one by one? Thanks.

**Comment by smuzaffar** [comment]
@srimanob , I case of timeout `runTheMatrix.py` do not generate the summary so yes it is hard to findout which one fail. I look at the logs and it is workflow `140.56`  which started at 9h26 and did not finish in 4 hours.

**Comment by srimanob** [comment]
Thanks @smuzaffar 
140.56 is the HI real data, we should not care about it with the modifier. Since it applies to all workflows during PR test, we may need to use commands with does not effect much on all of them.

**Comment by smuzaffar** [comment]
@srimanob , this PR itself is good to go in. This shows that we can now run PR  tests with extra `command "option"`. If needed, you can reset the `test parameters` and re-run the tests to get `test-approved` label

**Comment by srimanob** [comment]
test parameters:

**Comment by srimanob** [comment]
Please test

**Comment by tvami** [comment]
> @boudoul RelVal-INPUT issue is related to #34633
> As far as I can see, there is not yet a solution coded in a PR...

Hi @boudoul in the meantime the following PR was proposed to fix the time-out issue: https://github.com/cms-sw/cmssw/pull/34779

**Comment by srimanob** [comment]
+Upgrade

**Comment by chayanit** [comment]
+1

**Comment by perrotta** [comment]
+1
